wire-ios-sync-engine/Source/UserSession/ZMUserSession.swift

```diff
@@ -550,6 +550,17 @@ public class ZMUserSession: NSObject {
         }
     }
 
+    private func observeOnEpochChangeIfNeeded() {
+        guard e2eiFeature.isEnabled,
+              let mlsService = self.syncContext.mlsService else {
+            return
+        }
+        let epochChangeToken = EpochChangeObserver(
+            mlsService: mlsService,
+            block: self.mlsConversationVerificationStatusProvider?.updateStatus(_:))
```

Comment:
When `mlsConversationVerificationStatusProvider?.updateStatus(_:)` turn async this won't work. I propose to expose an `AsynsSequence` of epoch changes on the `MLSService`. See `onConferenceInfoChange`, it would then look something like:

```
func startObservingMLSConversationVerificationStatus() {
   Task {
      for groupID in mlsService.epochChanges {
         mlsConversationVerificationStatusProviderupdateStatus(groupID)
      }
   }
}

Reply:
Added